src/gradualizer_prelude_parse_trans.erl

```diff
@@ -35,4 +36,21 @@ get_module_and_forms(File) ->
     {ok, Forms} = epp:parse_file(File, []),
     [{attribute, _, file, _},
      {attribute, _, module, Module} | _] = Forms,
-    {Module, Forms}.
+    {Module, [ erl_anno_mark(F) || F <- Forms ]}.
+
+erl_anno_mark({eof, _} = Form) ->
+    Form;
+erl_anno_mark(Form) ->
+    Anno = element(2, Form),
+    setelement(2, Form, {'$anno', Anno}).
```

Comment:
What's the point of this? I don't understand.

Reply:
The point is that the annotation is just an integer or a pair of integers. After transforming into the `erl_syntax` representation it would be very hard to tell in `erl_syntax_lib:map()` mapper which integer (or which pair) that we see in the tree is an annotation and which is not. By leaving a marker like this one, we can then quite safely match on a tuple like `{tree, tuple, _, [{atom, _, '$anno'}, Anno]}` and finally substitute it with what we ultimately need here - an `erl_anno:new()` call. Why do we need that call? See the commit message on https://github.com/josefs/Gradualizer/pull/504/commits/9a5bd852f060f706f42d1e01a14a0a18bc48af2e.